pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -57,11 +58,16 @@ type message struct {
 // provider contains dependencies for the DID exchange protocol and is typically created by using aries.Context()
 type provider interface {
 	OutboundDispatcher() dispatcher.Outbound
+	Wallet
+}
+//TODO will be replaced by issue-280
+type Wallet interface {
+	NewDidDoc() *did.Doc
 }
-
 // Service for DID exchange protocol
 type Service struct {
 	outboundDispatcher dispatcher.Outbound
+	newDidDoc          *did.Doc
```

Comment:
The Service should be storing a handle to the Wallet interface such that a *new* `did.Doc` is created per execution of the did-exchange protocol

Reply:
done